src/argparse.cpp

```diff
@@ -60,8 +79,76 @@ bool ArgParse::parsingFails(Input& input, Output& output)
             return true;
         }
     }
+    // try to parse more flags after the positional arguments
+    while (!input.empty()) {
+        // only consume tokens if they are recognized flags
+        if (tryParseFlag(input.front())) {
+            input.shift();
+        } else {
+            break;
+        }
+    }
+
     // if all arguments were parsed, then we report that there were
     // no errors. It's ok if there are remaining elements in 'input' that
     // have not been parsed.
     return false;
 }
+
+/**
+ * @brief run parsingFails() and assert that there are no unknown
+ * tokens left in the input.
+ * @param input
+ * @param output
+ * @return whether there has been a
+ */
+bool ArgParse::parsingAllFails(Input& input, Output& output)
+{
+    return parsingFails(input, output) || unparsedTokens(input, output);
+}
+
+bool ArgParse::unparsedTokens(Input& input, Output& output)
+{
+    string extraToken;
+    if (input >> extraToken) {
+        output << input.command()
+           << ": Unknown argument or flag \""
+           << extraToken << "\" given.\n";
+        errorCode_ = HERBST_INVALID_ARGUMENT;
+        return true;
+    }
+    return false;
+}
+
+/**
+ * @brief Accept boolean flags (e.g. --all --horizontal ...) at
+ * any position between the (mandatory or optional) positional arguments
+ * @param a list of flags
+ * @return
```

Comment:
Left empty on purpose?

Reply:
Yes. Should I put `this` here?

Reply:
No, wouldn't add much value IMHO. Leave it is as is :)